dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -719,27 +840,34 @@ protected Ds3Metadata call() throws Exception {
                 final Ds3Client client = session.getClient();
                 final Ds3TreeTableValue value = values.get(0).getValue();
                 final HeadObjectResponse headObjectResponse = client.headObject(new HeadObjectRequest(value.getBucketName(), value.getFullName()));
-                return new Ds3Metadata(headObjectResponse.getMetadata(), headObjectResponse.getObjectSize(), value.getFullName());
+                return new Ds3Metadata(headObjectResponse.getMetadata(), headObjectResponse.getObjectSize(), value.getFullName(), value.getLastModified());
             }
         };
         workers.execute(getMetadata);
         getMetadata.setOnSucceeded(event -> Platform.runLater(() -> {
             LOG.info("Launching metadata popup");
-            MetadataPopup.show(getMetadata.getValue());
+            final MetadataView metadataView = new MetadataView(getMetadata.getValue());
+            Popup.show(metadataView.getView(), resourceBundle.getString("metaDataContextMenu"));
         }));
     }
 
     private void deleteFolderPrompt() {
         LOG.info("Got delete folder event");
-        final ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTable.getSelectionModel().getSelectedItems().stream().collect(GuavaCollectors.immutableList());
-        if (values.isEmpty()) {
+        ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTable.getSelectionModel().getSelectedItems().stream().collect(GuavaCollectors.immutableList());
```

Comment:
This logic seems to be repeated in several places.  Specifically the logic that is used to get the list of files that are selected, and if none are, providing a default of root.  Please attempt to consolidate this into a single method.